### PR TITLE
feat: US138562 - Update existing d2l-questions-multi-select and consume API

### PR DESCRIPTION
**Pull request by tdevos-d2l**
[US138562](https://rally1.rallydev.com/#/?detail=/userstory/632750608707&fdp=true): [Consistent Evaluation][Quizzes][UI][Questions]  Update existing d2l-questions-multi-select and hook up component to consume updated API

- Updated multi-select to consume API
- Updated read-only design
- Added unit tests
- Added v-diff tests
- Added aXe tests
- Added new demo data
- Added skeleton
- Updated docs

PR for Evaluating Method in CE coming after this is merged

### Read-only correct
image
### Read-only incorrect
image
### Answerable
image

### CODE REVIEW

File: components/d2l-questions-multi-select.js [hunk 1]
```diff
@@ -39,20 +59,78 @@ class D2lQuestionsMultiSelect extends SkeletonMixin(LitElement) {
...
+	async _loadChoices() {
+		// Reponse has the choices more readily available than the actual question does + the answer/correctness states
+		if (this.questionResponse) {
+			return await this._loadChoicesFromResponse();
+		}
+		const itemBodyHref = this.question.entity.getSubEntityByRel(Rels.Questions.itemBody);
+		const itemBodyEntity = await this._getEntityFromHref(itemBodyHref);
+		const interactionHref = itemBodyEntity.entity.getSubEntityByRel(Rels.Questions.interaction);
+		const interactionEntity = await this._getEntityFromHref(interactionHref);
+		const choices = await Promise.all(interactionEntity.entity.getSubEntitiesByClass(Classes.questions.simpleChoice).map(async choice => {
```
Comment (jeroach) [inline components/d2l-questions-multi-select.js:75]: Do we need null checks on any of these chained calls? Or are they all guaranteed to have a value?

File: components/d2l-questions-multi-select.js [hunk 2]
```diff
@@ -39,20 +59,78 @@ class D2lQuestionsMultiSelect extends SkeletonMixin(LitElement) {
...
+
+		this._choices = choices === undefined ? [] : choices;
+		return;
+	}
+
+	async _loadCorrectChoices(candidateResponse) {
+		const responseHref = candidateResponse.getLinkByRel(Rels.Questions.responseDeclaration).href;
+		const response = await this._getEntityFromHref(responseHref, false);
+		const correctResponses = response.entity.getSubEntityByClass(Classes.questions.correctResponse);
+		const correctChoiceEnitites = correctResponses.getSubEntitiesByClass(Classes.questions.value);
```
Comment (jeroach) [inline components/d2l-questions-multi-select.js:113]: Same question as above re: null checks and chained calls :)

File: components/d2l-questions-multi-select.js (hunk 1 above)
Comment (tdevos-d2l) [reply components/d2l-questions-multi-select.js:75]: Woops! Good point. I'll add some

**Review by jeroach — COMMENTED** [review]
Great work on all the test coverage!

I haven't been in this code much as of late, so others are more qualified for review purposes, but this looks solid overall to me.

Just a few questions around the potential need for null checks.

**Comment by tdevos-d2l** [comment]
> Great work on all the test coverage!
> 
> I haven't been in this code much as of late, so others are more qualified for review purposes, but this looks solid overall to me.
> 
> Just a few questions around the potential need for null checks.

Yeah I was thinking of waiting for Scy or Maddie to review before merging. I think I'll do that anyways since this isn't urgent

**Review by jeroach — APPROVED** [review]

**Review by JamieTran — APPROVED** [review]
Looks great overall! Love the test coverage 💯

**Review by maddiegabriel — APPROVED** [review]
great stuff Tristan!! LGTM :shipit:

**Pull request merged**